eoc_journal/eoc_journal.py

```diff
@@ -2,6 +2,12 @@
 An XBlock that allows learners to download their activity after they finish their course.
 """
 
+from collections import defaultdict
+
+from opaque_keys.edx.keys import CourseKey
+from problem_builder.models import Answer
+from problem_builder.step import _normalize_id
+from student.models import anonymous_id_for_user
```

Comment:
This will fail when not running inside an edX environment. While this particular xblock will be rather useless outside of edX, we should still make sure that it at least loads outside of edX (for example in the workbench, which we use for tests).

I think the best thing to do here is to duplicate the code from `problem-builder`: https://github.com/open-craft/problem-builder/blob/27e2f4ba5afdefcb6a4e5fd793969e1e626a0abe/problem_builder/answer.py#L60-L65